### ISSUE TITLE
^c doesn't interrupt/break at a prompt

### ISSUE THREAD

**Issue by IsaacG** [Kiwi/clyde#34]
When prompted for a [Y/n] ctrl-c is trapped and will not kill the application.

I strongly feel ^c should be allowed to kill the application.

**Comment by Kiwi** [comment]
Hit enter.

The signal catching is done by debug hooks and they only happen when a Lua function is called. This is why when alpm is doing something you cannot kill it immediately (like installing a package), but when it goes to the next package it will exit.

Not a bug.

**Comment by wraithan** [comment]
It is a bug, when you hit 1000+ search results and you want to cancle you need to be able to cancle out. By not allowing for C-c you are ignoring what 99% of other CLI programs do on linux.

**Comment by Kiwi** [comment]
Have you tried it? 1. you barely have time to hit ctrl+c before it is done anyway. 2. it actually does exit right away if you manage to.

Also, let me rephrase. Without completely changing, and not in a good way (think dark voodoo magic that opens a huge can of worms) signal handling (actually I am not even sure how/if it is possible) this behavior cannot be changed, even if I wanted to. Which I really do not.

For instance, if you are in the middle of installing or removing a package, and you ctrl+c, there is a strong chance you just messed something up really badly if it exits immediately. If, on the other hand, it finishes and then moves on to the next in the queue, but exits because it caught a signal, then you are a lot better off.

**Comment by wraithan** [comment]
Your app, not going to argue, just saying you are breaking the status quo for applications on linux. And in cases like the install process, it is C-c to break when it can, C-c C-c to break now regardless.

**Comment by Kiwi** [comment]
Even if it is the wrong way, it is not really something within my power to fix. I have sought guidance from those more experienced than me for alternate signal handling methods, including the author of the signal library I use, and basically the consensus is "Even if you can do it, you shouldn't"

**Comment by OttoAllmendinger** [comment]
The prompt at least is not a libalpm function, it is luas io.read() which cannot be interrupted by ctrl-c. 

The solution would be to use readline or some other means to get the user input

I think this is worth fixing, it is way easier to abort with ctrl-c than with ctrl-c + enter
